src/serial.rs

```diff
@@ -107,6 +107,67 @@ impl Pins<USART3> for (PB10<Alternate<PushPull>>, PB11<Input<Floating>>) {
 //     const REMAP: u8 = 0b11;
 // }
 
+pub enum Parity {
+    ParityNone,
+    ParityEven,
+    ParityOdd,
+}
+
+pub enum StopBits {
+    #[doc = "1 stop bit"]
+    STOP1,
+    #[doc = "0.5 stop bits"]
+    STOP0P5,
+    #[doc = "2 stop bits"]
+    STOP2,
+    #[doc = "1.5 stop bits"]
+    STOP1P5,
+}
+
+pub struct Config {
+    pub baudrate: Bps,
+    pub parity: Parity,
+    pub stopbits: StopBits,
+}
+
+impl Config {
+    pub fn baudrate(mut self, baudrate: Bps) -> Self {
+        self.baudrate = baudrate;
+        self
+    }
+
+    pub fn parity_none(mut self) -> Self {
+        self.parity = Parity::ParityNone;
+        self
+    }
+
+    pub fn parity_even(mut self) -> Self {
+        self.parity = Parity::ParityEven;
+        self
+    }
+
+    pub fn parity_odd(mut self) -> Self {
+        self.parity = Parity::ParityOdd;
+        self
+    }
+
+    pub fn stopbits(mut self, stopbits: StopBits) -> Self {
+        self.stopbits = stopbits;
+        self
+    }
+}
+
+impl Default for Config {
+    fn default() -> Config {
+        let baudrate = 19_200_u32.bps();
```

Comment:
19200 baud is kind of an odd choice, I haven't seen that in common use since the age of the acoustic coupler. Maybe either use 9600 or 115200 baud instead which are the two common default baud rates used nowadays (with a slight preference for the latter).

Reply:
You're absolutely right, I have no idea what made me chose 19200. 115200 makes sense to me.